File: src/api/map.ts
// HomeLens AI - 지도 API 호출 함수
// GET /api/v1/map/markers
// GET /api/v1/map/price-layer

import { apiGet } from "./client";

export interface MarkerItem {
  markerId: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
  markerType: string;
  avgPrice: number | null;
  priceLevel: string | null;
  distanceM: number | null;
}

export interface MapMarkerResponse {
  markers: MarkerItem[];
}

export interface PriceLayerZone {
  zoneId: string;
  lat: number;
  lng: number;
  value: number;
  priceGrade: number;
}

export interface PriceLayerResponse {
  zones: PriceLayerZone[];
  dataBaseDate: string;
}

// 지도 마커 조회 (아파트 + 인프라 마커)
export async function getMapMarkers(
  regionId: string,
  lat: number,
  lng: number,
  type: string = "all",
  infraRadius: number = 1500,
): Promise<MapMarkerResponse> {
  return apiGet<MapMarkerResponse>("/map/markers", {
    regionId,
    lat,
    lng,
    type,
    infraRadius,
  });
}

// 메인 지도 가격 레이어 조회
export async function getPriceLayer(
  regionId: string,
  type: string = "sale_count",
): Promise<PriceLayerResponse> {
  return apiGet<PriceLayerResponse>("/map/price-layer", {
    regionId,
    type,
  });
}

File: backend/app/schemas/map.py
# HomeLens AI - 지도 마커/레이어 API 응답 데이터 형식

from pydantic import BaseModel
from typing import Optional
from datetime import date

# 지도 마커 단건 데이터 형식
# apartment/subway/school/mart/hospital 마커 포함
class MarkerResponse(BaseModel):
    markerId: str
    name: str
    address: str
    lat: float
    lng: float
    markerType: str
    avgPrice: Optional[int] = None
    priceLevel: Optional[str] = None
    distanceM: Optional[int] = None

# 지도 마커 목록 응답 형식
# GET /api/v1/map/markers 응답에 사용
class MapMarkerResponse(BaseModel):
    markers: list[MarkerResponse]

# 가격 레이어 구역 단건 데이터 형식
class PriceLayerZone(BaseModel):
    zoneId: str
    lat: float
    lng: float
    value: float
    priceGrade: int
    aptName: Optional[str] = None
    aptSeq: Optional[str] = None
    kakaoPlaceId: Optional[str] = None

# 가격 레이어 응답 형식
# GET /api/v1/map/price-layer 응답에 사용
class PriceLayerResponse(BaseModel):
    zones: list[PriceLayerZone]
    dataBaseDate: date